android build error in next: Mesh.h:55:25: error: 'GL_READ_ONLY' was not declared in this scope

[armeabi-v7a] Compile++ arm  : gameplay <= AbsoluteLayout.cpp
In file included from jni/../../src/SpriteBatch.h:6:0,
                 from jni/../../src/Font.h:4,
                 from jni/../../src/Theme.h:6,
                 from jni/../../src/Control.h:7,
                 from jni/../../src/AbsoluteLayout.cpp:2:
jni/../../src/Mesh.h:55:25: error: 'GL_READ_ONLY' was not declared in this scope
         MAP_READ_ONLY = GL_READ_ONLY,
                         ^
jni/../../src/Mesh.h:56:26: error: 'GL_WRITE_ONLY' was not declared in this scope
         MAP_WRITE_ONLY = GL_WRITE_ONLY,
                          ^
jni/../../src/Mesh.h:57:26: error: 'GL_READ_WRITE' was not declared in this scope
         MAP_READ_WRITE = GL_READ_WRITE
                          ^
make: *** [obj/local/armeabi-v7a/objs/gameplay/AbsoluteLayout.o] Error 1

--- a/gameplay/src/Base.h
+++ b/gameplay/src/Base.h
@@ -217,6 +217,9 @@ using std::va_list;
     extern PFNGLDELETEVERTEXARRAYSOESPROC glDeleteVertexArrays;
     extern PFNGLGENVERTEXARRAYSOESPROC glGenVertexArrays;
     extern PFNGLISVERTEXARRAYOESPROC glIsVertexArray;
+    extern PFNGLMAPBUFFEROESPROC glMapBuffer;
+    extern PFNGLUNMAPBUFFEROESPROC glUnmapBuffer;
+    #define GL_WRITE_ONLY GL_WRITE_ONLY_OES
     #define GL_DEPTH24_STENCIL8 GL_DEPTH24_STENCIL8_OES
     #define glClearDepth glClearDepthf
     #define OPENGL_ES
@@ -239,6 +242,9 @@ using std::va_list;
         #define glDeleteVertexArrays glDeleteVertexArraysOES
         #define glGenVertexArrays glGenVertexArraysOES
         #define glIsVertexArray glIsVertexArrayOES
+        #define glMapBuffer glMapBufferOES
+        #define glUnmapBuffer glUnmapBufferOES
+        #define GL_WRITE_ONLY GL_WRITE_ONLY_OES
         #define GL_DEPTH24_STENCIL8 GL_DEPTH24_STENCIL8_OES
         #define glClearDepth glClearDepthf
         #define OPENGL_ES

--- a/gameplay/src/Mesh.cpp
+++ b/gameplay/src/Mesh.cpp
@@ -261,11 +261,11 @@ void Mesh::setPrimitiveType(PrimitiveType type)
     _primitiveType = type;
 }
 
-void* Mesh::mapVertexBuffer(MapAccess access)
+void* Mesh::mapVertexBuffer()
 {
     GL_ASSERT( glBindBuffer(GL_ARRAY_BUFFER, _vertexBuffer) );
 
-    return (void*)glMapBuffer(GL_ARRAY_BUFFER, access);
+    return (void*)glMapBuffer(GL_ARRAY_BUFFER, GL_WRITE_ONLY);
 }
 
 bool Mesh::unmapVertexBuffer()

--- a/gameplay/src/Mesh.h
+++ b/gameplay/src/Mesh.h
@@ -48,16 +48,6 @@ public:
     };
 
     /**
-     * Defines mapping access/usage.
-     */
-    enum MapAccess
-    {
-        MAP_READ_ONLY = GL_READ_ONLY,
-        MAP_WRITE_ONLY = GL_WRITE_ONLY,
-        MAP_READ_WRITE = GL_READ_WRITE
-    };
-
-    /**
      * Constructs a new mesh with the specified vertex format.
      *
      * @param vertexFormat The vertex format.
@@ -231,10 +221,9 @@ public:
      * vertex buffer become corrupted while the buffer was mapped. The corruption results from screen
      * resolution change or window system specific events. In this case, the data must be resubmitted.
      *
-     * @param access The access for which the data can be use. Ex. read, write, read_write.
      * @return The mapped vertex buffer
      */
-    void* mapVertexBuffer(Mesh::MapAccess access);
+    void* mapVertexBuffer();
 
     /**
      * Unmaps the vertex buffer.

--- a/gameplay/src/MeshPart.cpp
+++ b/gameplay/src/MeshPart.cpp
@@ -82,11 +82,11 @@ IndexBufferHandle MeshPart::getIndexBuffer() const
     return _indexBuffer;
 }
 
-void* MeshPart::mapIndexBuffer(Mesh::MapAccess access)
+void* MeshPart::mapIndexBuffer()
 {
     GL_ASSERT( glBindBuffer(GL_ELEMENT_ARRAY_BUFFER, _indexBuffer) );
 
-    return (void*)glMapBuffer(GL_ELEMENT_ARRAY_BUFFER, access);
+    return (void*)glMapBuffer(GL_ELEMENT_ARRAY_BUFFER, GL_WRITE_ONLY);
 }
 
 bool MeshPart::unmapIndexBuffer()

--- a/gameplay/src/MeshPart.h
+++ b/gameplay/src/MeshPart.h
@@ -78,10 +78,9 @@ public:
      * index buffer become corrupted while the buffer was mapped. The corruption results from screen
      * resolution change or window system specific events. In this case, the data must be resubmitted.
      *
-     * @param access The access for which the data can be use. Ex. read, write, read_write.
      * @return The mapped index buffer
      */
-    void* mapIndexBuffer(Mesh::MapAccess access);
+    void* mapIndexBuffer();
 
     /**
      * Unmaps the index buffer.

--- a/gameplay/src/PlatformAndroid.cpp
+++ b/gameplay/src/PlatformAndroid.cpp
@@ -52,6 +52,8 @@ PFNGLBINDVERTEXARRAYOESPROC glBindVertexArray = NULL;
 PFNGLDELETEVERTEXARRAYSOESPROC glDeleteVertexArrays = NULL;
 PFNGLGENVERTEXARRAYSOESPROC glGenVertexArrays = NULL;
 PFNGLISVERTEXARRAYOESPROC glIsVertexArray = NULL;
+PFNGLMAPBUFFEROESPROC glMapBuffer = NULL;
+PFNGLUNMAPBUFFEROESPROC glUnmapBuffer = NULL;
 
 #define GESTURE_TAP_DURATION_MAX			200
 #define GESTURE_LONG_TAP_DURATION_MIN   	GESTURE_TAP_DURATION_MAX
@@ -329,6 +331,8 @@ static bool initEGL()
         glDeleteVertexArrays = (PFNGLDELETEVERTEXARRAYSOESPROC)eglGetProcAddress("glDeleteVertexArraysOES");
         glGenVertexArrays = (PFNGLGENVERTEXARRAYSOESPROC)eglGetProcAddress("glGenVertexArraysOES");
         glIsVertexArray = (PFNGLISVERTEXARRAYOESPROC)eglGetProcAddress("glIsVertexArrayOES");
+        glMapBuffer = (PFNGLMAPBUFFEROESPROC)eglGetProcAddress("glMapBufferOES");
+        glUnmapBuffer = (PFNGLUNMAPBUFFEROESPROC)eglGetProcAddress("glUnmapBufferOES");
     }
     
     return true;

